src/ensembl/utils/archive.py

```diff
@@ -14,6 +14,8 @@
 # limitations under the License.
 """Utils for common IO operations over archive files, e.g. tar or gzip."""
 
+from __future__ import annotations
```

Comment:
This is advised in the type hint documentation to fully support newer versions of Python when writing type hints based on "older" versions, i.e. using Python 3.8 type hints but wanting to be fully compatible with 3.12.

Reply:
Is it a mandatory recommendation? I don't see the point of having this systematicallt in all our scripts if at least mypy doesn't complain

Reply:
It is not mandatory, just to support postponed evaluation of annotations. Since this code is meant to be maintained by more people, I felt there was no harm to leave it there for now.

Based on [this post](https://stackoverflow.com/questions/61544854/from-future-import-annotations), it will become part of Python at 3.11, so at that point that line can be removed.So based on [this post](https://stackoverflow.com/questions/61544854/from-future-import-annotations), it will become part of Python at 3.11, so at that point that line can be removed.